### ISSUE TITLE
Audit usage of header elements in Learner Dashboard

### ISSUE THREAD

**Issue by MaxFrank13** [openedx/frontend-app-learner-dashboard#531]
Headers (ie. `h1, h2, h3,...` tags) should be used carefully as they are really important for making pages accessible.

## Resources
[Confluence page on how to use headers](https://openedx.atlassian.net/l/cp/V4xPft0M)

## Multiple uses of an `h1` tag
There should only be one `h1` tag per page. Having more will potentially confuse a11y tools.

The usage that seems correct at first glance is in the Dashboard component where a pageTitle is rendered as an h1 ([source](https://github.com/openedx/frontend-app-learner-dashboard/blob/master/src/containers/Dashboard/index.jsx#L24)). This appears to be correct but we should verify.

There are at least two other places where an h1 is used:
- [NoCoursesView](https://github.com/openedx/frontend-app-learner-dashboard/blob/master/src/containers/CoursesPanel/NoCoursesView/index.jsx#L22-L24)
- [ConfirmEmailBanner](https://github.com/openedx/frontend-app-learner-dashboard/blob/master/src/containers/LearnerDashboardHeader/ConfirmEmailBanner/index.jsx#L67)

We’ll want to change these according to where they are in the layout.

## `h2` being used as a spacer
An `h2` element is being used to create white space in the layout for Learner Dashboard ([source](https://github.com/openedx/frontend-app-learner-dashboard/blob/a174abbc097637518f191c6f6f4cd919b396032a/src/containers/CoursesPanel/index.jsx#L32)):

<img width="609" alt="Screenshot 2024-09-26 at 8 46 01 AM" src="https://github.com/user-attachments/assets/7ccd6e52-55ee-40e9-8ae7-4f86df0d0886" />

Setting the element to a div doesn’t immediately fix the issue:

<img width="620" alt="Screenshot 2024-09-26 at 8 47 25 AM" src="https://github.com/user-attachments/assets/e89772d0-7241-4bc6-8d7f-a9f1b31fa7bd" />

There are custom styles being applied to the `course-list-title` class that are actually controlling the layout. 

This is a problem because `h2` should never be used for this purpose. There is no screen reader text or anything to indicate that this is used for accessibility in any way. 

We should try to find a way to achieve the same layout/view/look without using an `h2` for the spacing.

## Other usage of headers

We should verify that all other uses of header tags are up to accessibility standards and being used properly according to the [doc](https://openedx.atlassian.net/l/cp/V4xPft0M). 

## Acceptance Criteria

- There is only one `h1` and it follows best practice for what an `h1` should be (see [doc](https://openedx.atlassian.net/l/cp/V4xPft0M))
- `h2` is no longer used for spacing the layout
- Assess other uses of header elements
- If they are trivial, fix them
- If not, make follow up tickets as necessary
- The layout looks the same as before with no visual or behavioral changes in desktop and mobile views.
- A commit mentions the improvement for screen reader experiences

**Comment by navyanihitha** [comment]
Hi @MaxFrank13, I'm interested to work on this, could this issue be assigned to me.

**Comment by MaxFrank13** [comment]
Absolutely! Thanks for picking it up

**Comment by diana-villalvazo-wgu** [comment]
Hi @MaxFrank13 its been a while on this one and i don't see any opened PR, can i take it?

**Comment by diana-villalvazo-wgu** [comment]
Hi @openedx/2u-aperture ⬆ same as prev comment

**Comment by jsnwesson** [comment]
@diana-villalvazo-wgu feel free to pick this up, thank you!
